<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>NCBI BLAST • SPblast</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="NCBI BLAST">
<meta property="og:description" content="SPblast">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-CCMXLYXWYV"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CCMXLYXWYV');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPblast</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SPblast.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/SPblast/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>NCBI BLAST</h1>
                        <h4 data-toc-skip class="author">Author:
FirstName LastName</h4>
            
            <h4 data-toc-skip class="date">Last update: 02 October,
2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/SPblast/blob/HEAD/vignettes/SPblast.Rmd" class="external-link"><code>vignettes/SPblast.Rmd</code></a></small>
      <div class="hidden name"><code>SPblast.Rmd</code></div>

    </div>

    
    
<!--
Config css and r style
-->
<style type="text/css">
pre code {
white-space: pre !important;
overflow-x: scroll !important;
word-break: keep-all !important;
word-wrap: initial !important;
}
</style>
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><div class="section level2">
<h2 id="about-the-template">About the template<a class="anchor" aria-label="anchor" href="#about-the-template"></a>
</h2>
<p>This section provides general description and how to use this
cheminformatics workflow. In the actual analysis report, this section is
usually <strong>removed</strong>.</p>
<p>This BLAST workflow template is based on the <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="external-link">BLAST</a> based R
package <a href="https://github.com/mhahsler/rBLAST" class="external-link">rBLAST</a>.</p>
<ul>
<li>The BLAST software can be downloaded from <a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/" class="external-link">NCBI</a>.
Please make sure it can be run from command-line.</li>
<li>rBLAST can be installed with
<code>install.packages('rBLAST', repos = 'https://mhahsler.r-universe.dev')</code>.</li>
</ul>
<p>This workflow does: 1. Validate the BLAST installation 2. BLAST input
fasta file with a reference genome. 3. BLAST input with a certain
database 4. BLAST sequence with general databases to find out the source
organism(s).</p>
<p>All are written in R (<code>Linewise</code>) steps, but
<em>BLAST+</em> must be installed.</p>
<p><img src="results/plotwf_spblast.jpg"></p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Users want to provide here background information about the design of
their cheminformatics project.</p>
<p>This report describes the analysis of a BLAST project studying drug
…</p>
<div class="section level3">
<h3 id="experimental-design">Experimental design<a class="anchor" aria-label="anchor" href="#experimental-design"></a>
</h3>
<p>Typically, users want to specify here all information relevant for
the analysis of their BLAST study. This includes detailed descriptions
of files, experimental design, reference genome, gene annotations,
etc.</p>
</div>
</div>
<div class="section level2">
<h2 id="workflow-environment">Workflow environment<a class="anchor" aria-label="anchor" href="#workflow-environment"></a>
</h2>
<p><em><code>systemPipeR</code></em> workflows can be designed and built
from start to finish with a single command, importing from an R Markdown
file or stepwise in interactive mode from the R console.</p>
<p>This tutorial will demonstrate how to build the workflow in an
interactive mode, appending each step. The workflow is constructed by
connecting each step via <code>appendStep</code> method. Each
<code>SYSargsList</code> instance contains instructions needed for
processing a set of input files with a specific command-line or R
software and the paths to the corresponding outfiles generated by a
particular tool/step.</p>
<p>To create a Workflow within <em><code>systemPipeR</code></em>, we can
start by defining an empty container and checking the directory
structure:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/" class="external-link">systemPipeR</a></span><span class="op">)</span></span>
<span><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SPRproject.html" class="external-link">SPRproject</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">sal</span></span></code></pre></div>
<div class="section level3">
<h3 id="load-packages">Load packages<a class="anchor" aria-label="anchor" href="#load-packages"></a>
</h3>
<p>This is an empty template that contains only one demo step. Refer to
our <a href="https://systempipe.org/sp/spr/spr_run/" class="external-link">website</a> for how
to add more steps. If you prefer a more enriched template, <a href="https://systempipe.org/sp/spr/templates/" class="external-link">read this page</a> for
other pre-configured templates.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu">crayon</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/crayon/man/crayon.html" class="external-link">blue</a></span><span class="op">$</span><span class="fu">bold</span><span class="op">(</span><span class="st">"To use this workflow, following R packages are expected:\n"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"'rBLAST"</span>, <span class="st">"readr\n"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"', '"</span><span class="op">)</span></span>
<span><span class="co">### pre-end</span></span>
<span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/" class="external-link">systemPipeR</a></span><span class="op">)</span></span>
<span>    <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rBLAST</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"load_packages"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="test-blast-install">Test BLAST install<a class="anchor" aria-label="anchor" href="#test-blast-install"></a>
</h3>
<p>Molecules can be loaded or downloaded. This example dataset has 100
molecules.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Here, the dataset is downloaded. If you already have the</span></span>
<span><span class="co"># data locally, change URL to local path.</span></span>
<span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="co"># If you have a modular system, use following line</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/moduleload.html" class="external-link">moduleload</a></span><span class="op">(</span><span class="st">"ncbi-blast"</span><span class="op">)</span></span>
<span>    <span class="co"># If not, comment out line above you need to install</span></span>
<span>    <span class="co"># BLAST and configure the PATH.</span></span>
<span>    <span class="va">blast_check</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/tryCMD.html" class="external-link">tryCMD</a></span><span class="op">(</span><span class="st">"blastn"</span>, silent <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="va">blast_check</span> <span class="op">==</span> <span class="st">"error"</span><span class="op">)</span></span>
<span>        <span class="kw"><a href="https://rdrr.io/r/base/stop.html" class="external-link">stop</a></span><span class="op">(</span><span class="st">"Check your BLAST installation path."</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"test_blast"</span>, dependency <span class="op">=</span> <span class="st">"load_packages"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="load-query-sequence">Load query sequence<a class="anchor" aria-label="anchor" href="#load-query-sequence"></a>
</h3>
<p>Load query sequence from a <code>fasta</code> file.</p>
<p>In this template, an example fasta is provided, with 10 sequences
from Arabidopsis, Cholera, Human, Mouse, and COVID-19, 2 for each.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">query</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readDNAStringSet</a></span><span class="op">(</span><span class="st">"data/example.fasta"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"load_query"</span>, dependency <span class="op">=</span> <span class="st">"test_blast"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="blast-against-reference-genome">BLAST against reference genome<a class="anchor" aria-label="anchor" href="#blast-against-reference-genome"></a>
</h3>
<p>In this step, we are trying to BLAST the query sequences to a
reference genome and see if this genome contains the whole or part of
the sequences.</p>
<p>In this example, a minimized <code>tair10</code> genome is used. In
the real analysis, please replace it with a full genome
<code>fasta</code> file.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">reference</span> <span class="op">&lt;-</span> <span class="st">"data/tair10.fasta"</span></span>
<span>    <span class="co"># this command prepare BLAST-able database of genome</span></span>
<span>    <span class="fu">makeblastdb</span><span class="op">(</span><span class="va">reference</span>, dbtype <span class="op">=</span> <span class="st">"nucl"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"build_genome_db"</span>, dependency <span class="op">=</span> <span class="st">"load_query"</span><span class="op">)</span></span></code></pre></div>
<p>Next BLAST is performed. Since there are only 2 Arabidopsis sequences
in the example <code>fasta</code>. Only these two sequences are expected
to return statistically meaningful BLAST results.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">bl_tair10</span> <span class="op">&lt;-</span> <span class="fu">blast</span><span class="op">(</span>db <span class="op">=</span> <span class="va">reference</span>, type <span class="op">=</span> <span class="st">"blastn"</span><span class="op">)</span></span>
<span>    <span class="va">cl_tair10</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">bl_tair10</span>, <span class="va">query</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_csv</a></span><span class="op">(</span><span class="va">cl_tair10</span>, <span class="st">"results/blast_tair10.csv"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"blast_genome"</span>, dependency <span class="op">=</span> <span class="st">"build_genome_db"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="blast-existing-databases">BLAST existing databases<a class="anchor" aria-label="anchor" href="#blast-existing-databases"></a>
</h3>
<p>There are plenty of databases on <a href="https://ftp.ncbi.nlm.nih.gov/blast/db/" class="external-link">NCBI</a> that one could
download and run BLAST on. Once the databases are downloaded, unzip all
files into one directory. We need to provide the path to the
database.</p>
<p>In this example, we want to know if COVID-19 is a beta coronavirus.
Then, we can use some COVID sequence to BLAST all other existing beta
coronavirus sequences and find the similarity. This resource is
downloadable from NCBI. All downloaded
<code>Betacoronavirus.XX.tar.gz</code> files are unzipped to
<code>/srv/projects/db/ncbi/preformatted/20220131/</code>. Please change
the path according to your project. Then, we can BLAST the last two
sequence against the database.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">bl_covid</span> <span class="op">&lt;-</span> <span class="fu">blast</span><span class="op">(</span>db <span class="op">=</span> <span class="st">"/srv/projects/db/ncbi/preformatted/20220131/Betacoronavirus"</span>,</span>
<span>        type <span class="op">=</span> <span class="st">"blastn"</span><span class="op">)</span></span>
<span>    <span class="va">cl_covid</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">bl_covid</span>, <span class="va">query</span><span class="op">[</span><span class="fl">9</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_csv</a></span><span class="op">(</span><span class="va">cl_covid</span>, <span class="st">"results/blast_covid.csv"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"blast_db"</span>, dependency <span class="op">=</span> <span class="st">"load_query"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="blast-to-general-databases">BLAST to general databases<a class="anchor" aria-label="anchor" href="#blast-to-general-databases"></a>
</h3>
<p>Sometimes we do not know the origin of a sequence, for example, a
sequence comes from a contaminated sample, and we want to know the
source. In such cases, we would need to BLAST the sequence to a more
generic database. The most generic nucleotide BLAST database is the
<code>nt</code> database.</p>
<p>This database is extremely big and requires giant RAM and CPU cores
to run. Please do not run the following example unless your system admin
has provided you such store space and computational power. A better way
for average the user is to use the website <a href="https://blast.ncbi.nlm.nih.gov/blast/Blast.cgi/" class="external-link uri">https://blast.ncbi.nlm.nih.gov/blast/Blast.cgi/</a> . The
engine over there is optimized and can quickly search for the species
information.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="va">bl_nt</span> <span class="op">&lt;-</span> <span class="fu">blast</span><span class="op">(</span>db <span class="op">=</span> <span class="st">"/srv/projects/db/ncbi/preformatted/20220131/nt"</span>,</span>
<span>        type <span class="op">=</span> <span class="st">"blastn"</span><span class="op">)</span></span>
<span>    <span class="va">cl_nt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/predict.html" class="external-link">predict</a></span><span class="op">(</span><span class="va">bl_nt</span>, <span class="va">query</span><span class="op">[</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="fu">readr</span><span class="fu">::</span><span class="fu"><a href="https://readr.tidyverse.org/reference/write_delim.html" class="external-link">write_csv</a></span><span class="op">(</span><span class="va">cl_nt</span>, <span class="st">"results/blast_nt.csv"</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"blast_nt"</span>, dependency <span class="op">=</span> <span class="st">"load_query"</span>, run_step <span class="op">=</span> <span class="st">"optional"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="workflow-session">Workflow session<a class="anchor" aria-label="anchor" href="#workflow-session"></a>
</h3>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/LineWise-class.html" class="external-link">LineWise</a></span><span class="op">(</span>code <span class="op">=</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">}</span>, step_name <span class="op">=</span> <span class="st">"wf_session"</span>, dependency <span class="op">=</span> <span class="st">"blast_db"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="manage-the-workflow">Manage the workflow<a class="anchor" aria-label="anchor" href="#manage-the-workflow"></a>
</h2>
<p>To run the workflow, use <code>runWF</code> function. It executes all
the steps store in the workflow container. The execution will be on a
single machine without submitting to a queuing system of a computer
cluster.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runWF.html" class="external-link">runWF</a></span><span class="op">(</span><span class="va">sal</span>, run_step <span class="op">=</span> <span class="st">"mandatory"</span><span class="op">)</span>  <span class="co"># remove `run_step` to run all steps to include optional steps</span></span></code></pre></div>
<ul>
<li>To use complex workflow control options, such as parallelization,
subsetting samples, selecting steps, read the <a href="https://systempipe.org/sp/spr/sp_run/step_run/" class="external-link">documents</a> on
our website.</li>
<li>Explore <a href="https://systempipe.org/sp/spr/sp_run/sal_explore/" class="external-link">other details
of the workflow object</a>.</li>
<li>Create <a href="https://systempipe.org/sp/spr/sp_run/step_reports/" class="external-link">logs and
reports</a>.</li>
<li>
<a href="https://systempipe.org/sp/spr/sp_run/step_vis/" class="external-link">Visualize
the workflow</a>.</li>
</ul>
</div>
<div class="section level2">
<h2 id="about-the-workflow">About the workflow<a class="anchor" aria-label="anchor" href="#about-the-workflow"></a>
</h2>
<div class="section level3">
<h3 id="tools-used">Tools used<a class="anchor" aria-label="anchor" href="#tools-used"></a>
</h3>
<p>To check command-line tools used in this workflow, use
<code>listCmdTools</code>, and use <code>listCmdModules</code> to check
if you have a modular system.</p>
<p>The following code will print out tools required in your custom SPR
project in the report. In case you are running the workflow for the
first and do not have a project yet, or you just want to browser this
workflow, following code displays the tools required by default.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">".SPRproject"</span>, <span class="st">"SYSargsList.yml"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/eval.html" class="external-link">local</a></span><span class="op">(</span><span class="op">{</span></span>
<span>        <span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SPRproject.html" class="external-link">SPRproject</a></span><span class="op">(</span>resume <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>        <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/listCmdTools.html" class="external-link">listCmdTools</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></span>
<span>        <span class="fu">systemPipeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/listCmdTools.html" class="external-link">listCmdModules</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></span>
<span>    <span class="op">}</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu">crayon</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/crayon/man/crayon.html" class="external-link">blue</a></span><span class="op">$</span><span class="fu">bold</span><span class="op">(</span><span class="st">"Tools and modules required by this workflow are:\n"</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"bowtie2/2.4.5"</span>, <span class="st">"samtools/1.14"</span>, <span class="st">"macs2"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #0000BB; font-weight: bold;">Tools and modules required by this workflow are:</span></span></span>
<span><span class="co"><span style="color: #0000BB; font-weight: bold;">## </span>bowtie2/2.4.5</span></span>
<span><span class="co">## samtools/1.14</span></span>
<span><span class="co">## macs2</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h3>
<p>This is the session information for rendering this report. To access
the session information of workflow running, check HTML report of
<code>renderLogs</code>.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.3.1 (2023-06-16)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.6 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/Los_Angeles</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets </span></span>
<span><span class="co">## [6] methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BiocStyle_2.28.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] crayon_1.5.2          vctrs_0.6.2          </span></span>
<span><span class="co">##  [3] cli_3.6.1             knitr_1.42           </span></span>
<span><span class="co">##  [5] rlang_1.1.1           xfun_0.39            </span></span>
<span><span class="co">##  [7] stringi_1.7.12        purrr_1.0.1          </span></span>
<span><span class="co">##  [9] textshaping_0.3.6     jsonlite_1.8.4       </span></span>
<span><span class="co">## [11] glue_1.6.2            rprojroot_2.0.3      </span></span>
<span><span class="co">## [13] htmltools_0.5.5       formatR_1.14         </span></span>
<span><span class="co">## [15] ragg_1.2.5            sass_0.4.5           </span></span>
<span><span class="co">## [17] rmarkdown_2.21        evaluate_0.20        </span></span>
<span><span class="co">## [19] jquerylib_0.1.4       fastmap_1.1.1        </span></span>
<span><span class="co">## [21] lifecycle_1.0.3       yaml_2.3.7           </span></span>
<span><span class="co">## [23] memoise_2.0.1         bookdown_0.33        </span></span>
<span><span class="co">## [25] BiocManager_1.30.21.1 stringr_1.5.0        </span></span>
<span><span class="co">## [27] compiler_4.3.1        codetools_0.2-19     </span></span>
<span><span class="co">## [29] fs_1.6.2              rstudioapi_0.14      </span></span>
<span><span class="co">## [31] systemfonts_1.0.4     digest_0.6.31        </span></span>
<span><span class="co">## [33] R6_2.5.1              magrittr_2.0.3       </span></span>
<span><span class="co">## [35] bslib_0.4.2           tools_4.3.1          </span></span>
<span><span class="co">## [37] pkgdown_2.0.7         cachem_1.0.8         </span></span>
<span><span class="co">## [39] desc_1.4.2</span></span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Le Zhang, <a href="http://girke.bioinformatics.ucr.edu/" class="external-link">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
