<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>BLAST Workflow Template • SPblast</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="BLAST Workflow Template">
<meta property="og:description" content="SPblast">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=G-CCMXLYXWYV"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-CCMXLYXWYV');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">SPblast</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/SPblast.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/systemPipeR/SPblast/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="SPblast_files/htmlwidgets-1.5.4/htmlwidgets.js"></script><link href="SPblast_files/vizjs-2.1.2/plotwf.css" rel="stylesheet">
<script src="SPblast_files/vizjs-2.1.2/viz.js"></script><script src="SPblast_files/vizjs-2.1.2/full.render.js"></script><script src="SPblast_files/dom_to_image-2.6/dom_to_image.js"></script><link id="plotwf_legend-1-attachment" rel="attachment" href="SPblast_files/plotwf_legend-1/plotwf_legend.svg">
<script src="SPblast_files/plotwf-binding-2.1.9/plotwf.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>BLAST Workflow Template</h1>
                        <h4 data-toc-skip class="author">Author: Daniela Cassol (<a href="mailto:danielac@ucr.edu" class="email">danielac@ucr.edu</a>) and Thomas Girke (<a href="mailto:thomas.girke@ucr.edu" class="email">thomas.girke@ucr.edu</a>)</h4>
            
            <h4 data-toc-skip class="date">Last update: 06 December, 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/systemPipeR/SPblast/blob/HEAD/vignettes/SPblast.Rmd" class="external-link"><code>vignettes/SPblast.Rmd</code></a></small>
      <div class="hidden name"><code>SPblast.Rmd</code></div>

    </div>

    
    
<div class="section level1">
<h1 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h1>
<p>The <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="external-link">Basic Local Alignment Search Tool (BLAST)</a> finds regions of local similarity between sequences.</p>
<table class="table">
<colgroup>
<col width="12%">
<col width="12%">
<col width="15%">
<col width="61%">
</colgroup>
<thead><tr class="header">
<th>BLAST Tool</th>
<th>Query Type</th>
<th>Database Type</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>blastn</code></td>
<td>Nucleotide</td>
<td>Nucleotide</td>
<td>Search a nucleotide database using a nucleotide query</td>
</tr>
<tr class="even">
<td><code>blastx</code></td>
<td>Nucleotide</td>
<td>Protein</td>
<td>Search protein database using a translated nucleotide query</td>
</tr>
<tr class="odd">
<td><code>tblastn</code></td>
<td>Protein</td>
<td>Nucleotide</td>
<td>Search translated nucleotide database using a protein query</td>
</tr>
<tr class="even">
<td><code>blastp</code></td>
<td>Protein</td>
<td>Protein</td>
<td>Search protein database using a protein query</td>
</tr>
</tbody>
</table>
<div class="section level2">
<h2 id="experimental-design">Experimental design<a class="anchor" aria-label="anchor" href="#experimental-design"></a>
</h2>
<p>Typically, users want to specify here all information relevant for the analysis of their NGS study. This includes detailed descriptions of FASTQ files, experimental design, reference genome, gene annotations, etc.</p>
</div>
</div>
<div class="section level1">
<h1 id="workflow-environment">Workflow environment<a class="anchor" aria-label="anchor" href="#workflow-environment"></a>
</h1>
<div class="section level2">
<h2 id="load-packages-and-sample-data">Load packages and sample data<a class="anchor" aria-label="anchor" href="#load-packages-and-sample-data"></a>
</h2>
<p>The <code>systemPipeR</code> package needs to be loaded to perform the analysis steps shown in this report <span class="citation">(H Backman and Girke 2016)</span>. The package allows users to run the entire analysis workflow interactively or with a single command while also generating the corresponding analysis report. For details see <code>systemPipeR's</code> main <a href="http://www.bioconductor.org/packages/devel/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html" class="external-link">vignette</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://systempipe.org/" class="external-link">systemPipeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/systemPipeR/SPblast" class="external-link">SPblast</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="setup-and-requirements">Setup and Requirements<a class="anchor" aria-label="anchor" href="#setup-and-requirements"></a>
</h2>
<p>To go through this tutorial, you need the following software installed:</p>
<ul>
<li>R/&gt;=4.0.3</li>
<li>systemPipeR R package (version 1.24)</li>
<li>blastp: 2.9.0+</li>
</ul>
<p>If you desire to build your pipeline with any different software, make sure to have the respective software installed and configured in your PATH. To make sure if the configuration is right, you always can test as follow:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/tryCMD.html" class="external-link">tryCL</a></span><span class="op">(</span>command <span class="op">=</span> <span class="st">"blastp"</span><span class="op">)</span>  </code></pre></div>
<pre><code><span class="co">## [1] "All set up, proceed!"</span></code></pre>
</div>
<div class="section level2">
<h2 id="generate-workflow-environment">Generate workflow environment<a class="anchor" aria-label="anchor" href="#generate-workflow-environment"></a>
</h2>
<p><a href="http://bioconductor.org/packages/release/data/experiment/html/systemPipeRdata.html" class="external-link"><em>systemPipeRdata</em></a> package is a helper package to generate a fully populated <a href="http://bioconductor.org/packages/release/bioc/html/systemPipeR.html" class="external-link"><em>systemPipeR</em></a> workflow environment in the current working directory with a single command. All the instruction for generating the workflow template are provide in the <em>systemPipeRdata</em> vignette <a href="http://www.bioconductor.org/packages/devel/data/experiment/vignettes/systemPipeRdata/inst/doc/systemPipeRdata.html#1_Introduction" class="external-link">here</a>.</p>
<p>After building and loading the workflow environment generated by <code>genWorkenvir</code> from <em>systemPipeRdata</em> all data inputs are stored in a <code>data/</code> directory and all analysis results will be written to a separate <code>results/</code> directory, while the <code>SPblast.Rmd</code> script and the <code>targets</code> file are expected to be located in the parent directory. The R session is expected to run from this parent directory. Additional parameter files are stored under <code>param/</code>.</p>
<p>To work with real data, users want to organize their own data similarly and substitute all test data for their own data. To rerun an established workflow on new data, the initial <code>targets</code> file along with the corresponding FASTQ files are usually the only inputs the user needs to provide.</p>
<p>For more details, please consult the documentation <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html" class="external-link">here</a>. More information about the <code>targets</code> files from <em>systemPipeR</em> can be found <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/systemPipeR/inst/doc/systemPipeR.html#25_structure_of_targets_file" class="external-link">here</a>.</p>
</div>
<div class="section level2">
<h2 id="get-data">Get Data<a class="anchor" aria-label="anchor" href="#get-data"></a>
</h2>
<p>We are using for this tutorial mouse and human <a href="ftp://ftp.ncbi.nih.gov/refseq/">RefSeq</a> protein data sets from NCBI. First, we download the required dataset to <code>data/</code> folder and decompressing files, as follows:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">## add as step</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.1.protein.faa.gz"</span>, 
              destfile <span class="op">=</span> <span class="st">"data/mouse.1.protein.faa.gz"</span><span class="op">)</span>
<span class="fu">R.utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/R.utils/man/compressFile.html" class="external-link">gunzip</a></span><span class="op">(</span><span class="st">"data/mouse.1.protein.faa.gz"</span><span class="op">)</span>
<span class="co"># download.file("ftp://ftp.ncbi.nih.gov/refseq/M_musculus/mRNA_Prot/mouse.2.protein.faa.gz", </span>
<span class="co">#               destfile = "data/mouse.2.protein.faa.gz")</span>
<span class="co"># R.utils::gunzip("data/mouse.2.protein.faa.gz")</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="st">"ftp://ftp.ncbi.nih.gov/refseq/H_sapiens/mRNA_Prot/human.1.protein.faa.gz"</span>,
              destfile <span class="op">=</span> <span class="st">"data/human.1.protein.faa.gz"</span><span class="op">)</span>
<span class="fu">R.utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/R.utils/man/compressFile.html" class="external-link">gunzip</a></span><span class="op">(</span><span class="st">"data/human.1.protein.faa.gz"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/Biostrings" class="external-link">Biostrings</a></span><span class="op">)</span>
<span class="va">mouse.path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"data/mouse.1.protein.faa"</span>, package<span class="op">=</span><span class="st">"SPblast"</span><span class="op">)</span>
<span class="va">mouse.path</span> <span class="op">&lt;-</span> <span class="st">"data/mouse.1.protein.faa"</span>
<span class="va">select</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readAAStringSet</a></span><span class="op">(</span><span class="va">mouse.path</span><span class="op">)</span>
<span class="va">select</span> <span class="op">&lt;-</span> <span class="va">select</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">writeXStringSet</a></span><span class="op">(</span><span class="va">select</span>, <span class="st">"data/mouse_select.fasta"</span>, format<span class="op">=</span><span class="st">"fasta"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="project-initialization">Project initialization<a class="anchor" aria-label="anchor" href="#project-initialization"></a>
</h2>
<p>To create a Workflow within <em><code>systemPipeR</code></em>, we can start by defining an empty container and checking the directory structure:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SPRproject.html" class="external-link">SPRproject</a></span><span class="op">(</span><span class="op">)</span> </code></pre></div>
<pre><code><span class="co">## Creating directory: /home/dcassol/danielac@ucr.edu/projects/SP/SPR_org/Workflow/SPblast/SPblast/vignettes/data </span>
<span class="co">## Creating directory: /home/dcassol/danielac@ucr.edu/projects/SP/SPR_org/Workflow/SPblast/SPblast/vignettes/param </span>
<span class="co">## Creating directory: /home/dcassol/danielac@ucr.edu/projects/SP/SPR_org/Workflow/SPblast/SPblast/vignettes/results </span>
<span class="co">## Creating directory '/home/dcassol/danielac@ucr.edu/projects/SP/SPR_org/Workflow/SPblast/SPblast/vignettes/.SPRproject'</span>
<span class="co">## Creating file '/home/dcassol/danielac@ucr.edu/projects/SP/SPR_org/Workflow/SPblast/SPblast/vignettes/.SPRproject/SYSargsList.yml'</span></code></pre>
</div>
<div class="section level2">
<h2 id="create-workflow">Create workflow<a class="anchor" aria-label="anchor" href="#create-workflow"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargsList-class.html" class="external-link">SYSargsList</a></span><span class="op">(</span>step_name <span class="op">=</span> <span class="st">"makeblastdb"</span>, 
                               targets <span class="op">=</span> <span class="cn">NULL</span>, dir <span class="op">=</span> <span class="cn">FALSE</span>,
                               wf_file<span class="op">=</span><span class="st">"blast/makeblastdb.cwl"</span>, input_file<span class="op">=</span><span class="st">"blast/makeblastdb.yml"</span>,
                               dir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/cwl"</span>, package<span class="op">=</span><span class="st">"systemPipeR"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargsList-class.html" class="external-link">SYSargsList</a></span><span class="op">(</span>step_name <span class="op">=</span> <span class="st">"blastp"</span>, 
                               targets <span class="op">=</span> <span class="cn">NULL</span>, dir <span class="op">=</span> <span class="cn">TRUE</span>,
                               wf_file<span class="op">=</span><span class="st">"blast/blastp.cwl"</span>, input_file<span class="op">=</span><span class="st">"blast/blastp.yml"</span>,
                               dir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/cwl"</span>, package<span class="op">=</span><span class="st">"systemPipeR"</span><span class="op">)</span>, 
                               dependency <span class="op">=</span> <span class="st">"makeblastdb"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargsList-class.html" class="external-link">SYSargsList</a></span><span class="op">(</span>step_name <span class="op">=</span> <span class="st">"top_hits"</span>, 
                               targets <span class="op">=</span> <span class="cn">NULL</span>, dir <span class="op">=</span> <span class="cn">TRUE</span>,
                               wf_file<span class="op">=</span><span class="st">"blast/cat.cwl"</span>, input_file<span class="op">=</span><span class="st">"blast/cat.yml"</span>,
                               dir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/cwl"</span>, package<span class="op">=</span><span class="st">"systemPipeR"</span><span class="op">)</span>, 
                               dependency <span class="op">=</span> <span class="st">"blastp"</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">appendStep</span><span class="op">(</span><span class="va">sal</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/SYSargsList-class.html" class="external-link">SYSargsList</a></span><span class="op">(</span>step_name <span class="op">=</span> <span class="st">"blastdbcmd"</span>, 
                               targets <span class="op">=</span> <span class="cn">NULL</span>, dir <span class="op">=</span> <span class="cn">TRUE</span>,
                               wf_file<span class="op">=</span><span class="st">"blast/blastdbcmd.cwl"</span>, input_file<span class="op">=</span><span class="st">"blast/blastdbcmd.yml"</span>,
                               dir_path <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata/cwl"</span>, package<span class="op">=</span><span class="st">"systemPipeR"</span><span class="op">)</span>, 
                              dependency <span class="op">=</span> <span class="st">"top_hits"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="run-workflow">Run workflow<a class="anchor" aria-label="anchor" href="#run-workflow"></a>
</h2>
<p>For running the workflow, <code>runWF</code> function will execute all the command lines store in the workflow container.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">sal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/runWF.html" class="external-link">runWF</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="workflow-visualization">Workflow Visualization<a class="anchor" aria-label="anchor" href="#workflow-visualization"></a>
</h2>
<p>systemPipeR workflows instances can be visualized with the <code>plotWF</code> function.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/systemPipeR/man/plotWF.html" class="external-link">plotWF</a></span><span class="op">(</span><span class="va">sal</span><span class="op">)</span></code></pre></div>
<div id="htmlwidget-5b3b14b31fffe3d4b271" style="width:700px;height:432.632880098888px;" class="plotwf html-widget"></div>
<script type="application/json" data-for="htmlwidget-5b3b14b31fffe3d4b271">{"x":{"dot":"digraph {\n    node[fontsize=20];\n    subgraph {\n        makeblastdb -> blastp -> top_hits -> blastdbcmd\n   }\n    \n    \n    makeblastdb[fillcolor=\"#d3d6eb\" style=\"filled, rounded\" label=<<b><font color=\"black\">makeblastdb<\/font><br><\/br><font color=\"#5cb85c\">0<\/font>/<font color=\"#f0ad4e\">0<\/font>/<font color=\"#d9534f\">0<\/font>/<font color=\"blue\">11<\/font><\/b>; <font color=\"black\">0s<\/font>> , shape=\"box\"  tooltip=\"step makeblastdb: 0 samples passed; 0 samples have warnings; 0 samples have errors; 11 samples in total; Start time: 2021-12-06 09:10:53; End time: 2021-12-06 09:10:53; Duration: 00:00:00\"]\n    blastp[fillcolor=\"#d3d6eb\" style=\"filled, rounded\" label=<<b><font color=\"black\">blastp<\/font><br><\/br><font color=\"#5cb85c\">0<\/font>/<font color=\"#f0ad4e\">0<\/font>/<font color=\"#d9534f\">0<\/font>/<font color=\"blue\">1<\/font><\/b>; <font color=\"black\">0s<\/font>> , shape=\"box\"  tooltip=\"step blastp: 0 samples passed; 0 samples have warnings; 0 samples have errors; 1 samples in total; Start time: 2021-12-06 09:10:53; End time: 2021-12-06 09:10:53; Duration: 00:00:00\"]\n    top_hits[fillcolor=\"#d3d6eb\" style=\"filled, rounded\" label=<<b><font color=\"black\">top_hits<\/font><br><\/br><font color=\"#5cb85c\">0<\/font>/<font color=\"#f0ad4e\">0<\/font>/<font color=\"#d9534f\">0<\/font>/<font color=\"blue\">1<\/font><\/b>; <font color=\"black\">0s<\/font>> , shape=\"box\"  tooltip=\"step top_hits: 0 samples passed; 0 samples have warnings; 0 samples have errors; 1 samples in total; Start time: 2021-12-06 09:10:53; End time: 2021-12-06 09:10:53; Duration: 00:00:00\"]\n    blastdbcmd[fillcolor=\"#d3d6eb\" style=\"filled, rounded\" label=<<b><font color=\"black\">blastdbcmd<\/font><br><\/br><font color=\"#5cb85c\">0<\/font>/<font color=\"#f0ad4e\">0<\/font>/<font color=\"#d9534f\">0<\/font>/<font color=\"blue\">1<\/font><\/b>; <font color=\"black\">0s<\/font>> , shape=\"box\"  tooltip=\"step blastdbcmd: 0 samples passed; 0 samples have warnings; 0 samples have errors; 1 samples in total; Start time: 2021-12-06 09:10:53; End time: 2021-12-06 09:10:53; Duration: 00:00:00\"]\n        subgraph cluster_legend {\n        rankdir=TB;\n        color=\"#eeeeee\";\n        style=filled;\n        ranksep =1;\n        label=\"Legends\";\n        fontsize = 30;\n        node [style=filled, fontsize=10];\n        legend_img-> step_state[color=\"#eeeeee\"];\n\n        legend_img[shape=none, image=\"plotwf_legend-src.png\", label = \" \", height=1, width=3, style=\"\"];\n\n        step_state[style=\"filled\", shape=\"box\" color=white, label =<\n            <table>\n            <tr><td><b>Step Colors<\/b><\/td><\/tr>\n            <tr><td><font color=\"black\">Pending steps<\/font>; <font color=\"#5cb85c\">Successful steps<\/font>; <font color=\"#d9534f\">Failed steps<\/font><\/td><\/tr>\n            <tr><td><b>Targets Files / Code Chunk <\/b><\/td><\/tr><tr><td><font color=\"#5cb85c\">0 (pass) <\/font> | <font color=\"#f0ad4e\">0 (warning) <\/font> | <font color=\"#d9534f\">0 (error) <\/font> | <font color=\"blue\">0 (total)<\/font>; Duration<\/td><\/tr><\/table>\n            >];\n\n    }\n\n}\n","plotid":"sprwf-21738654","responsive":true,"width":null,"height":null,"plot_method":"renderSVGElement","rmd":true,"msg":"","plot_ctr":true,"pan_zoom":false},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div class="section level1">
<h1 id="version-information">Version Information<a class="anchor" aria-label="anchor" href="#version-information"></a>
</h1>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R Under development (unstable) (2021-10-25 r81105)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.3 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0</span>
<span class="co">## LAPACK: /home/dcassol/src/R-devel/lib/libRlapack.so</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] SPblast_0.99.0              systemPipeR_2.1.9          </span>
<span class="co">##  [3] ShortRead_1.53.1            GenomicAlignments_1.31.2   </span>
<span class="co">##  [5] SummarizedExperiment_1.25.2 Biobase_2.55.0             </span>
<span class="co">##  [7] MatrixGenerics_1.7.0        matrixStats_0.61.0         </span>
<span class="co">##  [9] BiocParallel_1.29.4         Rsamtools_2.11.0           </span>
<span class="co">## [11] Biostrings_2.63.0           XVector_0.35.0             </span>
<span class="co">## [13] GenomicRanges_1.47.5        GenomeInfoDb_1.31.1        </span>
<span class="co">## [15] IRanges_2.29.1              S4Vectors_0.33.5           </span>
<span class="co">## [17] BiocGenerics_0.41.2         BiocStyle_2.23.0           </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##  [1] sass_0.4.0             jsonlite_1.7.2         bslib_0.3.1           </span>
<span class="co">##  [4] assertthat_0.2.1       BiocManager_1.30.16    latticeExtra_0.6-29   </span>
<span class="co">##  [7] GenomeInfoDbData_1.2.7 yaml_2.2.1             pillar_1.6.4          </span>
<span class="co">## [10] lattice_0.20-45        glue_1.5.1             digest_0.6.29         </span>
<span class="co">## [13] RColorBrewer_1.1-2     colorspace_2.0-2       htmltools_0.5.2       </span>
<span class="co">## [16] Matrix_1.3-4           pkgconfig_2.0.3        bookdown_0.24         </span>
<span class="co">## [19] zlibbioc_1.41.0        purrr_0.3.4            scales_1.1.1          </span>
<span class="co">## [22] jpeg_0.1-9             tibble_3.1.6           generics_0.1.1        </span>
<span class="co">## [25] ggplot2_3.3.5          ellipsis_0.3.2         cachem_1.0.6          </span>
<span class="co">## [28] magrittr_2.0.1         crayon_1.4.2           memoise_2.0.1         </span>
<span class="co">## [31] evaluate_0.14          fs_1.5.1               fansi_0.5.0           </span>
<span class="co">## [34] hwriter_1.3.2          textshaping_0.3.6      tools_4.2.0           </span>
<span class="co">## [37] lifecycle_1.0.1        stringr_1.4.0          munsell_0.5.0         </span>
<span class="co">## [40] DelayedArray_0.21.2    compiler_4.2.0         pkgdown_2.0.1         </span>
<span class="co">## [43] jquerylib_0.1.4        systemfonts_1.0.3      rlang_0.4.12          </span>
<span class="co">## [46] grid_4.2.0             RCurl_1.98-1.5         htmlwidgets_1.5.4     </span>
<span class="co">## [49] bitops_1.0-7           rmarkdown_2.11         gtable_0.3.0          </span>
<span class="co">## [52] DBI_1.1.1              R6_2.5.1               knitr_1.36            </span>
<span class="co">## [55] dplyr_1.0.7            fastmap_1.1.0          utf8_1.2.2            </span>
<span class="co">## [58] rprojroot_2.0.2        ragg_1.2.1             desc_1.4.0            </span>
<span class="co">## [61] stringi_1.7.6          parallel_4.2.0         vctrs_0.3.8           </span>
<span class="co">## [64] png_0.1-7              tidyselect_1.1.1       xfun_0.28</span></code></pre>
</div>
<div class="section level1">
<h1 id="funding">Funding<a class="anchor" aria-label="anchor" href="#funding"></a>
</h1>
<p>This project is funded by NSF award ABI-1661152.</p>
</div>
<div class="section level1 unnumbered">
<h1 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h1>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-H_Backman2016-bt" class="csl-entry">
H Backman, Tyler W, and Thomas Girke. 2016. <span>“<span class="nocase">systemPipeR: NGS workflow and report generation environment</span>.”</span> <em>BMC Bioinformatics</em> 17 (1): 388. <a href="https://doi.org/10.1186/s12859-016-1241-0" class="external-link">https://doi.org/10.1186/s12859-016-1241-0</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by <a href="https://www.dcassol.com/" class="external-link">Daniela Cassol</a>, <a href="http://girke.bioinformatics.ucr.edu/" class="external-link">Thomas Girke</a>.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
